assets/dev/js/editor/regions/panel/pages/elements/elements.js

```diff
@@ -65,6 +65,14 @@ PanelElementsLayoutView = Marionette.LayoutView.extend( {
 		const elementsCollection = new PanelElementsElementsCollection(),
 			isContainerActive = elementorCommon.config.experimentalFeatures.container;
 
+		Object.entries( elementor.widgetsCache ).forEach( ( [ widgetName, widgetData ] ) => {
+			// If new version loaded.
```

Comment:
remove comment 

please write this code in a more clean way  :

the final code should look like 
Object.entries( elementor.widgetsCache ).forEach(hideOldTabWidgetsIfNeeded(paramsNeeded))

hideOldTabWidgetsIfNeeded should look like 
hideOldTabWidgetsIfNeeded (paramsNeeded){
  if (isNewTabWidgetLoaded){
     hideOldTabWidget();
   }
}

hideOldTabWidget  should look like 
hideOldTabWidget(paramsNeeded )
elementor.widgetsCache[ widgetName ].show_in_panel = false
]

Reply:
Hi @TzviRabinovitch it should be a generic code that will support other deprecated widgets

Reply:
@matipojo then write it generic 
you can add the widget name to hide or any other param in the call to the function 
clean code can be generic too :)